staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go

```diff
@@ -104,6 +119,14 @@ func (o *ExplainOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []
 		return err
 	}
 
+	if o.EnableOpenAPIV3 {
+		clientset, err := f.KubernetesClientSet()
+		if err != nil {
+			return err
+		}
+		o.DiscoveryClient = clientset.DiscoveryClient
```

Comment:
You mean by passing the openapi v3 client directly instead of the schema? I did not consider there to be tradeoffs. I think it helps code organization a bit.

Reply:
It is not possible to know which schema to fetch until `Run` is called and the GVR is parsed, etc.

I could move that code up here but I think that'd be a bigger change?

@seans3  can you advise on whether we should fetch the schema here in `Complete` would require moving https://github.com/kubernetes/kubernetes/blob/bac82250d595716a0de80ba245be01a7728845d2/staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go#L157-L173

otherwise I think storing discovery client is the only way